web/api/v1/api_test.go

```diff
@@ -191,6 +191,12 @@ func (t testAlertmanagerRetriever) DroppedAlertmanagers() []*url.URL {
 	}
 }
 
+func (t testAlertmanagerRetriever) toFactory() func(context.Context) AlertmanagerRetriever {
```

Comment:
I would vote to just inline, it's just used once....

Reply:
less LOC

Reply:
`gofmt` only allowed inlining of the inner function on this one, did that.